cherry_picker/cherry_picker/cherry_picker.py

```diff
@@ -72,10 +73,16 @@ def upstream(self):
 
     @property
     def sorted_branches(self):
+        def version_from_branch(branch):
+            try:
+                return tuple(map(int, re.match(r'^.*(?P<version>\d+(\.\d+)+).*$', branch).groupdict()['version'].split('.')))
```

Comment:
Maybe reading regex template from config is even better?

Reply:
I see how it's more flexible. I was thinking about this, but IRL regex is easy to screw up: they'll be exposed to the internals and helpless, and it would require them to actually scan through the code to figure out how it's being used, why and how to fix that. It also looks like mixing up higher-level public API with implementation details, which IMHO should be separated. So I'd be cautious when providing the end-users with such tooling.

What do you think?

Reply:
Maybe invite a syntax like `stable-*` where the star is *version*?

Reply:
It seems to be more complex than needed